WooCommerce/Classes/ViewRelated/Upgrades/UpgradesViewModel.swift

```diff
@@ -0,0 +1,187 @@
+        }
+
+        let planName = getPlanName(from: plan)
+        guard let expireDate = plan.expiryDate else {
+            return ""
+        }
+
+        let expireText = DateFormatter.mediumLengthLocalizedDateFormatter.string(from: expireDate)
+        return Localization.planInfo(planName: planName, expirationDate: expireText)
+    }
+
+    /// Only allow to upgrade the plan if we are on a free trial.
+    ///
+    static func getUpgradeNowButtonVisibility(from plan: WPComSitePlan) -> Bool {
+        plan.isFreeTrial
+    }
+
+    /// Returns a site plan duration in days.
+    ///
+    static func planDurationInDays(for plan: WPComSitePlan) -> Int {
+        // Normalize dates in the same timezone.
+        guard let subscribedDate = plan.subscribedDate?.startOfDay(timezone: .current),
+              let expiryDate = plan.expiryDate?.startOfDay(timezone: .current) else {
+            return 0
+        }
+
+        let duration = Calendar.current.dateComponents([.day], from: subscribedDate, to: expiryDate).day ?? 0
+        return duration
+    }
+
+    /// Returns how many days site  plan has left.
+    ///
+    static func daysLeft(for plan: WPComSitePlan) -> Int {
+        // Normalize dates in the same timezone.
+        let today = Date().startOfDay(timezone: .current)
+        guard let expiryDate = plan.expiryDate?.startOfDay(timezone: .current) else {
+            return 0
+        }
+
+        let daysLeft = Calendar.current.dateComponents([.day], from: today, to: expiryDate).day ?? 0
+        return daysLeft
+    }
+}
+
+// MARK: Definitions
+private extension UpgradesViewModel {
+    enum Localization {
+        static let freeTrial = NSLocalizedString("Free Trial", comment: "Plan name for an active free trial")
+        static let trialEnded = NSLocalizedString("Trial ended", comment: "Plan name for an expired free trial")
+        static let trialEndedInfo = NSLocalizedString("Your free trial has ended and have limited access to all the features. Subscribe to eCommerce now.",
+                                                      comment: "Info details for an expired free trial")
+        static func freeTrialPlanInfo(planDuration: Int, daysLeft: Int) -> String {
+            let format = NSLocalizedString("You are in the %1d-day free trial. The free trial will end in %2d days. " +
+                                           "Upgrade to unlock new features and keep your store running.",
+                                           comment: "Reads like: You are in the 14-day free trial. The free trial will end in 5 days. " +
+                                           "Upgrade to unlock new features and keep your store running.")
+            return String.localizedStringWithFormat(format, planDuration, daysLeft)
+        }
+
+        static func planInfo(planName: String, expirationDate: String) -> String {
+            let format = NSLocalizedString("You are a %1@ subscriber! You have access to all our features until %2@.",
```

Comment:
This string has a small problem in English: With plans that start with a vowel (like `eCommerce`) it should say "You are an eCommerce subscriber!" (`an` instead of `a`).

This is always a risk when individual words are inserted into strings like this, but at least in English it could be improved with an alternative like `You are subscribed to the %1@ plan!`.

It's not a huge deal if we don't change this, but maybe worth considering at some point.